src/main/java/com/segmentreader/domain/User.java

```diff
@@ -1,28 +1,43 @@
 package com.segmentreader.domain;
+
+import java.time.Instant;
 import java.util.LinkedList;
 import java.util.List;
+
 public class User {
-	String userId;
-	List<String> segments;
-	
-	public User(String userId, List<String> segments) {
-		super();
-		this.userId = userId;
-		this.segments = segments;
-	}
-	
-	public String getUserId() {
-		return userId;
-	}
-	public void setUserId(String userId) {
-		this.userId = userId;
-	}
-
-	public List<String> getSegments() {
-		return segments;
-	}
-
-	public void setSegments(LinkedList<String> segment) {
-		this.segments = segment;
-	}
+    Instant timestamp;
```

Comment:
I think this data structure should attach timestamp to segments list. otherwise it is absurd, how you will choose between multiple timestamps ;)